frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -71,17 +76,37 @@ export const generateClientKeySecret = (clientKey: string) => ({
   },
 });
 
-const generateCsiKmsConfigMap = (name: string, csiConfigData: KMSConfigMap) => ({
-  apiVersion: ConfigMapModel.apiVersion,
-  kind: ConfigMapModel.kind,
-  data: {
-    [`${name}`]: JSON.stringify(csiConfigData),
-  },
-  metadata: {
-    name: KMSConfigMapCSIName,
-    namespace: CEPH_STORAGE_NAMESPACE,
-  },
-});
+const convertCsiKeysToCamelCase = (csiConfigData: KMSConfigMap) => {
+  // ceph-csi supports only camelcase config keys for vault SA.
+  // And partially support uppser case for vault token.
+  // To avoid confusion, I am converting all vault config keys to camelcase.
+  const configData = csiConfigData as VaultSAConfigMap;
```

Comment:
aren't we converting keys to camelCase for all the KMS providers (Vault, VaultSA, HPCS) ? 
Why are we only typecasting it `as VaultSAConfigMap` ??

Reply:
I missed changing this place. But still, conversion is working fine :P LOL